snap7/util/db.py

```diff
@@ -84,7 +86,7 @@ class DB:
     """
 
     bytearray_: Optional[bytearray] = None  # data from plc
-    specification: Optional[str] = None  # layout of db rows
+    specification: str  # layout of db rows
```

Comment:
also this one should be optional i think

Reply:
Should it? `specification` is passed to DBRow in the constructor of the `DB` class, when it is passed to [`parse_specification`](https://github.com/gijzelaerr/python-snap7/blob/master/snap7/util/db.py#L351), which expects `str`. If you pass `None` to `parse_specification`, an exception will be raised. So I think `mypy` correctly indicates that None is not a suitable type.

![image](https://github.com/gijzelaerr/python-snap7/assets/52915342/6785bb04-d575-475d-871e-2534af621306)